CancerGov/_src/Scripts/NCI/Modules/customEventHandler/index.js

```diff
@@ -0,0 +1,111 @@
+        isCustomEventHandlerAttached = true;
+        __attachCustomEventHandler__();
+    }
+};
+export default initialize;
+
+/**
+ * DO NOT CALL DIRECTLY!
+ * This is a setup function that should not be used directly.
+ * It is only being exported for testing purposes.
+ */
+export const __attachCustomEventHandler__ = () => {
+    if(typeof window !== undefined){
+        const eventHandler = event => {
+            const { 
+                target, 
+                detail = {}
+            } = event;
+            const { 
+                eventType, 
+                data
+            } = detail;
+            if(
+                typeof eventType === 'string' 
+                && registeredEventListeners.hasOwnProperty(eventType) 
+                && Array.isArray(registeredEventListeners[eventType])
+            ){
+                const listeners = registeredEventListeners[eventType];
+                listeners.forEach(listener => listener(target, data));
+            }
+        };
+    
+        window.addEventListener(customEventGlobalNamespace, eventHandler);
+    }
+};
+
+/**
+ * Dispatch custom events on a DOM node on execution.
+ *  
+ * 
+ * @param {string} eventType Used by the customEventHandler to find the appropriate listener function to execute
+ * @param {object} settings
+ * @param {HTMLElement} settings.node The event target node
+ * @param {object} [settings.data = {}] Any custom user data to be passed through the CustomEvent detail object
+ * @return {function} Event Handler
+ */
+export const broadcastCustomEvent = createCustomEventBroadcaster(customEventGlobalNamespace);
+
+/**
+ * Register an event listener to the customEventHandler listener store.
+ * 
+ * @param {string} eventType the key used to reference the listener
+ * @param {function} listener
+ * @return {object} { eventType: string, listener: function }
+ */
+export const registerCustomEventListener = (eventType, listener) => {
+    if(typeof eventType !== 'string' && typeof listener !== 'function'){
+        throw new Error('Expected custom event listener to be a function')
+    }
+    
+    // If the eventType is already registered we want to add to the array, otherwise create a new array of listeners
+    registeredEventListeners = { 
+        ...registeredEventListeners, 
+        [eventType]: registeredEventListeners.hasOwnProperty(eventType) ? [ ...registeredEventListeners[eventType], listener ] : [ listener ]
+    };
+    return { eventType, listener };
+}
+
+/**
+ * Remove a custom listener from the customEventHandler listener store.
+ * 
+ * @param {string} eventType
+ * @return {object} { eventType: string, listener: function }
+ */
+export const unregisterCustomEventListener = (eventType, listenerToUnregister) => {
+    if(!registeredEventListeners.hasOwnProperty(eventType)){
+        return;
+    }
+
+    const listeners = registeredEventListeners[eventType];
+    if(Array.isArray(listeners) && listeners.length){
```

Comment:
When would this not be the case?  On line 77 [eventType] should always be set to an array.

Reply:
Haha. I actually thought you might say something about this little piece of obvious redundancy. :) It's called no-test CYA style. :)

But the fact that you pointed it out immediately speaks to my initial feelings about the redundancy.